lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,250 @@
+
+	configModule, err := createConfig(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate azure config module")
+	}
+
+	activityLogModule, err := createActivityLog(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate azure activity log module")
+	}
+
+	// Render
+	hclBlocks := lwgenerate.CreateHclStringOutput(
+		lwgenerate.CombineHclBlocks(
+			requiredProviders,
+			azureADProvider,
+			azureRMProvider,
+			laceworkProvider,
+			configModule,
+			activityLogModule),
+	)
+	return hclBlocks, nil
+}
+
+func createRequiredProviders() (*hclwrite.Block, error) {
+	return lwgenerate.CreateRequiredProviders(
+		lwgenerate.NewRequiredProvider(
+			"azuread",
+			lwgenerate.HclRequiredProviderWithSource(lwgenerate.HashAzureADProviderSource),
+			lwgenerate.HclRequiredProviderWithVersion(lwgenerate.HashAzureADProviderVersion),
+		),
+		lwgenerate.NewRequiredProvider(
+			"azurerm",
+			lwgenerate.HclRequiredProviderWithSource(lwgenerate.HashAzureRMProviderSource),
+			lwgenerate.HclRequiredProviderWithVersion(lwgenerate.HashAzureRMProviderVersion),
+		),
+	)
+}
+
+func createAzureADProvider() ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	//attrs := map[string]interface{}{}
+	provider, err := lwgenerate.NewProvider(
+		"azuread",
+	).ToBlock()
+
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, provider)
+	return blocks, nil
+}
+
+func createAzureRMProvider() ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	attrs := map[string]interface{}{}
+
+	provider, err := lwgenerate.NewProvider(
+		"azurerm",
+	).ToBlock()
+
+	if err != nil {
+		return nil, err
+	}
+	// Create the features block
+	featuresBlock, err := lwgenerate.HclCreateGenericBlock("features", []string{}, attrs)
+	provider.Body().AppendBlock(featuresBlock)
+
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, provider)
+	return blocks, nil
+}
+
+func createLaceworkAZADModule() ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+
+	provider, err := lwgenerate.NewModule(
+		"az_ad_application",
+		lwgenerate.LWAzureADSource,
+		lwgenerate.HclModuleWithVersion(lwgenerate.LWAzureADVersion),
+	).ToBlock()
+
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, provider)
+	return blocks, nil
+}
+
+func createConfig(args *GenerateAzureTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	if args.Config {
+		attributes := map[string]interface{}{}
+		moduleDetails := []lwgenerate.HclModuleModifier{}
+
+		if args.ConfigIntegrationName != "" {
+			attributes["lacework_integration_name"] = args.ConfigIntegrationName
+		}
+
+		// Set default values
+		attributes["use_existing_ad_application"] = true
+		attributes["application_id"] = "module.az_ad_application.application_id"
+		attributes["application_password"] = "module.az_ad_application.application_password"
+		attributes["service_principal_id"] = "module.az_ad_application.service_principal_id"
```

Comment:
I think for these we should be able to leave them blank if there are not being overridden, based on the module source: https://github.com/lacework/terraform-azure-config/blob/main/main.tf#L10-L16